installer/Migration/V5_4_0/Migration.php

```diff
@@ -135,19 +116,156 @@ public function up(): void
+            ],
+            'submitted_date' => [
+                'Type' => Type::getType(Types::DATETIME_MUTABLE),
+            ]
+        ]);
+
+        $this->getSchemaHelper()->addOrChangeColumns('ohrm_expense_type', [
+            'name' => [
+                'Notnull' => true,
+                'Type' => Type::getType(Types::STRING),
+            ],
+            'description' => [
+                'Type' => Type::getType(Types::STRING),
+                'Length' => 1000
+            ],
+            'is_deleted' => [
+                'Type' => Type::getType(Types::BOOLEAN),
+            ],
+            'status' => [
+                'Type' => Type::getType(Types::BOOLEAN),
+                'CustomSchemaOptions' => ['collation' => null,'charset' => null]
+            ]
+        ]);
+
+        $this->getSchemaHelper()->addOrChangeColumns('ohrm_claim_event', [
+            'name' => [
+                'Notnull' => true,
+                'Type' => Type::getType(Types::STRING),
+            ],
+            'description' => [
+                'Length' => 1000
+            ],
+            'is_deleted' => [
+                'Type' => Type::getType(Types::BOOLEAN),
+            ],
+        ]);
+
+        $this->getSchemaHelper()->renameColumn('ohrm_claim_request', 'currency', 'currency_id');
     }
+
+    private function modifyClaimRequestCurrencyToForeignKey(): void
+    {
+        $foreignKeyConstraint = new ForeignKeyConstraint(
+            ['currency_id'],
+            'hs_hr_currency_type',
+            ['currency_id'],
+            'fk_currency_id',
+            ['onDelete' => 'RESTRICT', 'onUpdate' => 'CASCADE']
+        );
+        $this->getSchemaHelper()->addForeignKey('ohrm_claim_request', $foreignKeyConstraint);
+    }
+
+    private function changeClaimExpenseTypeTableStatusToBoolean(): void
+    {
+        $table = 'ohrm_expense_type';
+        $count = $this->getTableRecordCount($table);
+        $batchSize = 10;
+        for ($i = 0; $i <= $count; $i = $i + $batchSize) {
+            $result = $this->createQueryBuilder()
+                ->select('expenseType.id', 'expenseType.status')
+                ->from($table, 'expenseType')
+                ->setFirstResult($i)
+                ->setMaxResults($batchSize)
+                ->executeQuery();
+
+            foreach ($result->fetchAllAssociative() as $row) {
+                $this->createQueryBuilder()
+                    ->update($table, 'expenseType')
+                    ->set('expenseType.status', ':status')
+                    ->where('expenseType.id = :id')
+                    ->setParameter('id', $row['id'])
+                    ->setParameter(
+                        'status',
+                        $this->setStatusToBoolean($row['status']),
+                        Types::BOOLEAN
+                    )
+                    ->executeStatement();
+            }
+            $result->free();
+        }
+    }
```

Comment:
@RajithaKumara need to check whether this is ok or not.

Reply:
two update queries are enough right?

1. on => 1
2. null => 0

Reply:
ya. I check there was no 'off' kind of thing. only 'on' and null